presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -649,24 +716,41 @@ public void fail(Throwable cause)
         stateMachine.updateQueryInfo(Optional.ofNullable(scheduler).map(SqlQuerySchedulerInterface::getStageInfo));
     }
 
+    /**
+     * Checks if the query is done executing
+     * @return true if the query is done executing
+     */
     @Override
     public boolean isDone()
     {
         return getState().isDone();
     }
 
+    /**
+     * Register a listener to be notified about new {@link QueryOutputInfo} buffers created as tasks execute in this query
+     * @param listener the listener
+     */
     @Override
     public void addOutputInfoListener(Consumer<QueryOutputInfo> listener)
     {
         stateMachine.addOutputInfoListener(listener);
     }
 
+    /**
+     * Gets a future that completes when the current query state is no longer equal to the current state
```

Comment:
is no longer equal sounds too much like code. Lets use "when the current query state has changed"